docs/building-a-form/about-the-json-schema-standard-and-react-jsonschema-form.md

```diff
@@ -0,0 +1,296 @@
+
+##### Describing object fields and arrays
+
+This example describes a JSON document that is an object with one property called `myField`, which is a number, meaning `{ myField: 2 }` would be valid:
+
+```
+{
+  type: 'object',
+  properties: {
+    myField: {
+      type: 'number'
+    }
+  }
+}
+```
+
+However, `{}` is also valid. To require a property in an object, use the `required` property. In this example, `required` is on the object that contains the field, not the field itself:
+
+```
+{
+  type: 'object',
+  required: ['myField'],
+  properties: {
+    myField: {
+      type: 'number'
+    }
+  }
+}
+```
+
+Arrays work similarly to objects. This example describes an array of boolean values: `[true, false, true]`. Items can be an object schema or any other type of schema as well:
+
+```
+{
+  type: 'array',
+  items: {
+    type: 'boolean'
+  }
+}
+```
+
+Many libraries implement the JSON Schema specification and let you validate that an object matches a given schema. The US Forms System uses [ajv](https://www.npmjs.com/package/ajv) for unit tests and [jsonschema](https://www.npmjs.com/package/jsonschema) in the schemaform code. For more examples of JSON Schema projects, see the [JSON Schema site example list](http://json-schema.org/examples.html).
+
+### Understanding react-jsonschema-form (rjsf)
+
+[react-jsonschema-form](https://github.com/mozilla-services/react-jsonschema-form), or *rjsf*, generates a form from a JSON Schema, in addition to other UI information. To generate a form, react-jsonschema-form steps through the schema depth and renders different React components based on the type of data each property in the schema represents.
+
+##### About the `Form` component
+
+At the top level, rjsf uses a `Form` component to take the schema inputs and render a hierarchy of components for each field rendered on the form:
+
+- *Fields* generally match the `type` attribute in a schema. There are object fields, array fields, number fields, boolean fields, and string fields. Except for arrays and objects, the fields render a label (via `FieldTemplate`) and a widget.
+- A *widget* is the html input element that accepts data from the user. The schemaform uses `text`, `email`, `checkbox`, `radio`, `select`, and `textarea`. While there are many widgets provided by rjsf, the defaults are overwritten with these versions.
+
+##### Example schema
+
+```
+{
+  type: 'string'
+}
+```
+
+##### Resulting hierarchy
+
+The two `Field` components determine which fields and widgets to render. `SchemaField` uses the two schemas the library accepts, `schema` and `uiSchema`, to determine what other `Field` component to render. The example chose `StringField` because the schema type was `string`. The `StringField` component then rendered `TextWidget`, based on `schema` and `uiSchema`, because the only information provided was that the field is a string (the default widget type).
```

Comment:
> `SchemaField` uses the two schemas the library accepts, `schema` and `uiSchema`

We've seen fragments of `schema` above but not `uiSchema`; did/would it have any effect on the rendering below? It would be handy to see a complete set of both schema that generated the result below.